### ISSUE TITLE
Add facet for sequencing_protocol.paired_end

### ISSUE THREAD

**Issue by theathorn** [DataBiosphere/data-browser#551]
@NoopDog commented on [Tue Apr 02 2019](https://github.com/HumanCellAtlas/data-portal/issues/399)

As a consumer of HCA data, I need to know whether a Smart-seq2 dataset is paired-end or not in order to  be sure that my workflow, which only analyzes paired-end SS2 data, will work.

As a consumer of HCA data, I need to be able to find only paired end data because I want to be able to have higher confidence in the results of my analysis, and paired end data offers higher coverage of transcripts than single end.

This will also help us signify if the data is processable by the matrix service.

- [x] Add to Method facet category
- [x] Add to the right of Library Construction Approach
- [x] Add to the same column as library construction approach in the tables as a 2nd line

**Comment by frano-m** [comment]
@dave 
1. Should we take libraryConstructionApproach from protocols rather than projectSummary
2. What is the expected outcome if the protocol array.length > 1?  Can this happen?
3. What do we do if pairedEnd is an array and can we associate the "ended-ness" with the protocol?

![image.png](https://images.zenhubusercontent.com/5ac463024b5806bc2bcd91dd/05244e86-e09c-42cb-94ef-311112c8d740)

**Comment by theathorn** [comment]
No linkage between SS2 + paired end or 10x + single end.
Need azul support from @hannes-ucsc - discuss.

**Comment by NoopDog** [comment]
The goal of this feature is to allow the user to determine what kind of library construction method and paired end *combinations* are used in a project. We want to see values like the following in the Library Construction Approach Column.

Paired End Smart Seq2
Single End Smart Seq2


The last instruction I got from Tim T for this was to display this only for the SmartSeq data. So what I can do is the following.

1. Add the facet

1. For smart Seq2 if there is only one value of Paired End, use that to append "Paired End" or "Single End" to the value in the Library Construction Method" column.  If there are multiple values of paired end e.g. [true,false] then do not display the Paired end  info in the Library Construction Method column (as we are not sure what the value is for a given Method)

1. for other than SS2 methods, ignore the value (as per Tim as 10x and others are not really single or paired end and the field may need a metadata API change to sort this out properly e.g change from boolean to enum )

I think this can get us over the current hump for this specific facet. 

@theathorn @hannes-ucsc WDYT?

**Comment by hannes-ucsc** [comment]
> The goal of this feature is to allow the user to determine what kind of library construction method and paired end combinations are used in a project.

I disagree. The goal of the feature is to expose an independent metadata property as an independent facet.

> I think this can get us over the current hump for this specific facet.

There is no hump as far as I can tell.

If the metadata team decided to expose single-endedness as a separate metadata property, independent from library construction method, the UI should reflect that. We must not hard-code business logic when it is based on current, potentially incidental observations such as an apparent correlation between single-endedness and library construction approach.

TL;DR: Let's just add the facet. We can then open this discussion up to a larger audience, pointing at a concrete implementation to demonstrate the suspected flaw.

**Comment by theathorn** [comment]
@briandoconnor agreed that browser should show paired_end as an independent facet rather than building in business logic to the data browser. For a project it should show all values that apply to the current selection (true/false/unassigned). So for SS2-only projects it would show "true" only and for 10x "false" and/or "unassigned", etc. This assumes that Azul surfaces the paired_end state correctly for all projects @hannes-ucsc.

**Comment by NoopDog** [comment]
Ok then, @frano-m lets go ahead and add the "Paired End" field to the right of the "Library Construction Method" field on all 3 tabs. 

Show all the values in the array.
Follow the normal rules for empty arrays.
Show the contents of the array without modification (e.g. true/false) except for the normal rules for the "Unspecified"

**Comment by NoopDog** [comment]
This is on dev https://dev.data.humancellatlas.org/explore/projects
@theathorn  for your review.

**Comment by theathorn** [comment]
When I select "Unspecified" as a filter value I get all the "true" projects.
Also some 10x projects have "true" values - is this correct?

**Comment by NoopDog** [comment]
Currently due to a a limitation in Azul, its not possible to select unspecified as a filter value. We should probably create a ticket for this if there is not one already. @theathorn 

Usually this gives no results so not sure why in dev it is listing all "true" values. @danielsotirhos do you have an idea why its behaving this way. 

 Maybe we should not allow the user to select "Unspecified" as a filter until this is properly handled by the app?

>Also some 10x projects have "true" values - is this correct

I think the 10x text data is inconsistent so this is just a data problem.

Perhaps we can cleanup the dev data sometime soon?

**Comment by hannes-ucsc** [comment]
Known issue: https://github.com/DataBiosphere/azul/issues/989

**Comment by NoopDog** [comment]
@theathorn can this be closed now? We have a reference to Azul 989 for the unspecified. 
Check this out in prod. I think its otherwise working ok.

**Comment by dsotirho-ucsc** [comment]
@NoopDog @hannes-ucsc There's something wonky going on with filtering by the pairedEnd facet, my guess is because it's a boolean value. Unsure if this will be addressed by azul#989

```
https://dev.data.humancellatlas.org/explore/samples?filter=%5B%7B%22facetName%22%3A%22pairedEnd%22%2C%22terms%22%3A%5B%22true%22%5D%7D%5D
URL-Decoded: filter=[{"facetName":"pairedEnd","terms":["true"]}]
25 pages of "true" paired end results (good)

https://dev.data.humancellatlas.org/explore/samples?filter=%5B%7B%22facetName%22%3A%22pairedEnd%22%2C%22terms%22%3A%5B%22false%22%5D%7D%5D
URL-Decoded: filter=[{"facetName":"pairedEnd","terms":["false"]}]
14 pages of "false" paired end results (good)

https://dev.data.humancellatlas.org/explore/samples?filter=%5B%7B%22facetName%22%3A%22pairedEnd%22%2C%22terms%22%3A%5B%22foo%22%5D%7D%5D
URL-Decoded: filter=[{"facetName":"pairedEnd","terms":["foo"]}]
25 pages of "true" paired end results (bad)

https://dev.data.humancellatlas.org/explore/samples?filter=%5B%7B%22facetName%22%3A%22pairedEnd%22%2C%22terms%22%3A%5B%22%22%5D%7D%5D
URL-Decoded: filter=[{"facetName":"pairedEnd","terms":[""]}]
14 pages of "false" paired end results (bad)
```

**Comment by theathorn** [comment]
@danielsotirhos Do we need a a separate ticket for "wonky"?

**Comment by dsotirho-ucsc** [comment]
> @danielsotirhos Do we need a a separate ticket for "wonky"?

No, this was just something I found and wanted to point out as to why a search for "Unspecified" returned the same results that a search for "true" did (because the field is a boolean and a search of any string of length 1+ is considered the same as a search for "true").
This issue should be addressed in https://github.com/DataBiosphere/azul/issues/989 or https://github.com/DataBiosphere/azul/issues/537
